src/main/java/org/openstreetmap/atlas/generator/AtlasGeneratorHelper.java

```diff
@@ -560,6 +561,45 @@ public String getName()
         };
     }
 
+    protected static PairFunction<Tuple2<String, Atlas>, String, Atlas> subatlas(
+            final ConfiguredFilter filter, final AtlasCutType cutType)
+    {
+        return tuple ->
+        {
+            final Atlas subAtlas;
+            // Grab the tuple contents
+            final String shardName = tuple._1();
+            final Atlas originalAtlas = tuple._2();
+            logger.info("Starting sub Atlas for for Atlas {}", originalAtlas.getName());
+            final Time start = Time.now();
+            try
+            {
+                // Slice the Atlas
+                final Optional<Atlas> subAtlasOptional = originalAtlas.subAtlas(filter::test,
+                        cutType);
+                if (subAtlasOptional.isPresent())
+                {
+                    subAtlas = subAtlasOptional.get();
+                }
+                else
+                {
+                    subAtlas = null;
+                    logger.error("Unable to extract valid subAtlas code for {}", shardName);
+                }
+            }
+            catch (final Throwable e) // NOSONAR
```

Comment:
In what case would this be a throwable and not an Exception?

Reply:
Good point, I'll change to Exception